### PR TITLE
Add Collection ID support

### PR DESCRIPTION
**Pull request by d4s**
Add support of Ref bindings (branches) and Collection ID.

Signed-off-by: Denis Pynkin <denis.pynkin@collabora.com>

### CODE REVIEW

File: pkg/otbuiltin/commit.go [hunk 1]
```diff
@@ -412,6 +415,29 @@ func parseKeyValueStrings(pairs []string) (*C.GVariant, error) {
 	builder := C.g_variant_builder_new(C._g_variant_type(C.CString("a{sv}")))
 	defer C.g_variant_builder_unref(builder)
 
+	// Add Ref bindings
+	sort.Strings(options.RefBindings)
```
Comment (sjoerdsimons) [inline pkg/otbuiltin/commit.go]: Eh this is a function to go from a generic keyvalue pair strings to a GVariant, It really shouldn't have this code added to it.. 

Was there a rebase accident here or ? Tbh i'm surprised you can access `options` here; but in any case if you need a less generic conversion function please make it that rather then doing a generic key value converstion and adding in a bunch of extra unexpected bits on the side :)

Comment (d4s) [reply pkg/otbuiltin/commit.go]: Hope new iteration looks better.

File: pkg/otbuiltin/commit.go [hunk 2]
```diff
@@ -407,6 +412,62 @@ out:
...
+		// C.OSTREE_COMMIT_META_KEY_REF_BINDING is available since ostree 2017.9
+	}
+
+	// Add Collection ID binding
+	if strings.Compare(collectionStr, "") != 0 {
+		collection_id := C.g_variant_new_string((*C.gchar)(C.CString(collectionStr)))
+		C._g_variant_builder_add_twoargs(builder, C.CString("{sv}"),
+			C.CString("ostree.collection-binding"),
+			collection_id)
+		// C.OSTREE_COMMIT_META_KEY_COLLECTION_BINDING is available since ostree 2018.6
```
Comment (sjoerdsimons) [inline pkg/otbuiltin/commit.go]: ditto; 2018.6 is new enough as well  (newer versions are available in Debian backports)

File: pkg/otbuiltin/commit.go [hunk 3]
```diff
@@ -407,6 +412,62 @@ out:
...
+	}
+
+	// Add previous metadata pairs
+	if unsafe.Pointer(metadata) != nil {
```
Comment (sjoerdsimons) [inline pkg/otbuiltin/commit.go]: Can you not add the previous metadata using g_value_builder_add_value at the start ? (potentially not)

Comment (d4s) [reply pkg/otbuiltin/commit.go]: I don't think so, tbh.

**Pull request merged**